Gemfile.lock

```diff
@@ -0,0 +1,171 @@
+      dry-logic (~> 1.0, >= 1.0.2)
+    ffi (1.15.4)
+    ffi-compiler (1.0.1)
+      ffi (>= 1.0.0)
+      rake
+    figaro (1.2.0)
+      thor (>= 0.14.0, < 2)
+    flog (4.6.4)
+      path_expander (~> 1.0)
+      ruby_parser (~> 3.1, > 3.1.0)
+      sexp_processor (~> 4.8)
+    hashdiff (1.0.1)
+    hirb (0.7.3)
+    hirb-unicode (0.0.5)
+      hirb (~> 0.5)
+      unicode-display_width (~> 0.1.1)
+    http (5.0.4)
+      addressable (~> 2.8)
+      http-cookie (~> 1.0)
+      http-form_data (~> 2.2)
+      llhttp-ffi (~> 0.4.0)
+    http-cookie (1.0.4)
+      domain_name (~> 0.5)
+    http-form_data (2.3.0)
+    ice_nine (0.11.2)
+    kwalify (0.7.2)
+    listen (3.7.0)
+      rb-fsevent (~> 0.10, >= 0.10.3)
+      rb-inotify (~> 0.9, >= 0.9.10)
+    llhttp-ffi (0.4.0)
+      ffi-compiler (~> 1.0)
+      rake (~> 13.0)
+    method_source (1.0.0)
+    mini_portile2 (2.6.1)
+    minitest (5.14.4)
+    minitest-rg (5.2.0)
+      minitest (~> 5.0)
+    nio4r (2.5.8)
+    nokogiri (1.12.5)
+      mini_portile2 (~> 2.6.1)
+      racc (~> 1.4)
+    nokogiri (1.12.5-x86_64-linux)
+      racc (~> 1.4)
+    parser (3.0.2.0)
+      ast (~> 2.4.1)
+    path_expander (1.1.0)
+    pg (1.2.3)
+    pry (0.13.1)
+      coderay (~> 1.1)
+      method_source (~> 1.0)
+    public_suffix (4.0.6)
+    puma (5.5.2)
+      nio4r (~> 2.0)
+    racc (1.6.0)
+    rack (2.2.3)
+    rainbow (3.0.0)
+    rake (13.0.6)
+    rb-fsevent (0.11.0)
+    rb-inotify (0.10.1)
+      ffi (~> 1.0)
+    rb-kqueue (0.2.7)
+      ffi (>= 0.5.0)
+    reek (6.0.6)
+      kwalify (~> 0.7.0)
+      parser (~> 3.0.0)
+      rainbow (>= 2.0, < 4.0)
+    rerun (0.13.1)
+      listen (~> 3.0)
+    rexml (3.2.5)
+    roda (3.50.0)
+      rack
+    rubocop (0.7.2)
+      rainbow (>= 1.1.4)
+    ruby_parser (3.18.1)
+      sexp_processor (~> 4.16)
+    sequel (5.50.0)
+    sexp_processor (4.16.0)
+    simplecov (0.21.2)
+      docile (~> 1.1)
+      simplecov-html (~> 0.11)
+      simplecov_json_formatter (~> 0.1)
+    simplecov-html (0.12.3)
+    simplecov_json_formatter (0.1.3)
+    slim (4.1.0)
+      temple (>= 0.7.6, < 0.9)
+      tilt (>= 2.0.6, < 2.1)
+    sqlite3 (1.4.2)
+    temple (0.8.2)
+    thor (1.1.0)
+    tilt (2.0.10)
+    unf (0.1.4)
+      unf_ext
+    unf_ext (0.0.8)
+    unicode-display_width (0.1.1)
+    vcr (6.0.0)
+    webmock (3.14.0)
+      addressable (>= 2.8.0)
+      crack (>= 0.3.2)
+      hashdiff (>= 0.4.0, < 2.0.0)
+
+PLATFORMS
+  ruby
+  x86_64-linux
+
+DEPENDENCIES
+  dry-struct (~> 1.4)
+  dry-types (~> 1.5)
+  figaro (~> 1.2)
+  flog
+  hirb (~> 0)
+  hirb-unicode (~> 0)
+  http (~> 5.0)
+  minitest (~> 5.0)
+  minitest-rg (~> 5.0)
+  nokogiri (~> 1.12)
+  pg
+  pry (~> 0.13.1)
+  puma (~> 5.5)
+  rake
+  rb-fsevent
+  rb-kqueue
+  reek
+  rerun
+  roda (~> 3.49)
+  rubocop
+  sequel (~> 5.5)
+  simplecov (~> 0)
+  slim (~> 4.1)
+  sqlite3 (~> 1.4)
+  vcr (~> 6.0)
+  webmock (~> 3.0)
+
+RUBY VERSION
+   ruby 3.0.2p107
+
+BUNDLED WITH
+   1.17.2
```

Comment:
Just curious about why the bundler version is so low? Any gems force you to use  version 1.17.2?

Reply:
I was using an older version of ruby before, and after upgrading ruby, an error occurs:
`Could not find 'bundler' (1.17.3)`

I did `bundle update --bundler` but no good,
so I tried `gem install bundler:1.17.3`

I was searching for a quick fix hence did not go deep into this issue

[Stackoverflow reference](https://stackoverflow.com/questions/57758654/why-do-i-get-could-not-find-bundler-1-17-3-after-upgrading-ruby)

Reply:
Oh I mean the bundler version, `bundle update --bundler` won't upgrade your bundler version, you should install a new version one. No need to change if it works fine. I just afraid some gems will conflict due to installing by older version bundler. If you want to update the bundler, you should type `gem install bundler -v 2.2.22` then `bundle update --bundler`. It depends on you to upgrade your bundler or not.